pinot-core/src/test/java/org/apache/pinot/queries/ArgMinMaxTest.java

```diff
@@ -0,0 +1,613 @@
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 4);
+
+    assertEquals(rows.get(0)[0], 7996000D);
+    assertEquals(rows.get(0)[1], 8D);
+    assertEquals(rows.get(0)[2], "a11");
+    assertEquals(rows.get(0)[3], 8D);
+
+    assertNull(rows.get(1)[0]);
+    assertEquals(rows.get(1)[1], 18D);
+    assertEquals(rows.get(1)[2], "a11");
+    assertEquals(rows.get(1)[3], 8D);
+
+    assertNull(rows.get(2)[0]);
+    assertEquals(rows.get(2)[1], 8D);
+    assertEquals(rows.get(2)[2], "a11");
+    assertNull(rows.get(2)[3]);
+
+    assertNull(rows.get(3)[0]);
+    assertEquals(rows.get(3)[1], 18D);
+    assertEquals(rows.get(3)[2], "a11");
+    assertNull(rows.get(3)[3]);
+
+    // Test transformation function inside argmax/argmin, for both projection and measuring
+    // the max of 3000x-x^2 is 2250000, which is the max of 3000x-x^2
+    query = "SELECT sum(intColumn), argmax(3000 * doubleColumn - intColumn * intColumn, doubleColumn),"
+        + "argmax(3000 * doubleColumn - intColumn * intColumn, 3000 * doubleColumn - intColumn * intColumn),"
+        + "argmax(3000 * doubleColumn - intColumn * intColumn, doubleColumn), "
+        + "argmin(replace(stringColumn, \'a\', \'bb\'), replace(stringColumn, \'a\', \'bb\'))"
+        + "FROM testTable";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 4);
+
+    assertEquals(rows.get(0)[0], 7996000D);
+    assertEquals(rows.get(0)[1], 1500D);
+    assertEquals(rows.get(0)[2], 2250000D);
+    assertEquals(rows.get(0)[3], "bb11");
+    assertNull(rows.get(1)[0]);
+    assertEquals(rows.get(1)[1], 1500D);
+    assertEquals(rows.get(1)[2], 2250000D);
+    assertEquals(rows.get(1)[3], "bb11");
+    assertNull(rows.get(2)[0]);
+    assertNull(rows.get(2)[1]);
+    assertEquals(rows.get(2)[3], "bb11");
+    assertNull(rows.get(3)[0]);
+    assertNull(rows.get(3)[1]);
+    assertEquals(rows.get(3)[3], "bb11");
+
+    // Inter segment mix aggregation function with CASE statement
+    query = "SELECT argmin(CASE WHEN stringColumn = 'a33' THEN 'b' WHEN stringColumn = 'a22' THEN 'a' ELSE 'c' END"
+        + ", stringColumn), argmin(CASE WHEN stringColumn = 'a33' THEN 'b' WHEN stringColumn = 'a22' THEN 'a' "
+        + "ELSE 'c' END, CASE WHEN stringColumn = 'a33' THEN 'b' WHEN stringColumn = 'a22' THEN 'a' ELSE 'c' END) "
+        + "FROM testTable";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 4);
+
+    for (int i = 0; i < 4; i++) {
+      assertEquals(rows.get(i)[0], "a22");
+      assertEquals(rows.get(i)[1], "a");
+    }
+
+    // TODO: The following query throws an exception,
+    //       requires fix for multi-value bytes column serialization in DataBlock
+    query = "SELECT arg_min(intColumn, mvBytesColumn) FROM testTable";
```

Comment:
Can we also add other failure scenario tests ? For example, invalid number or types of arguments in the `arg_min` or `arg_max` function if not already added ?

Reply:
Good point, added